torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -245,50 +305,39 @@ void insertQuantizationOps(
   }
   Value* original_val = observer->input(1);
   Node *quant, *choose_qparams, *dequant;
-  if (quant_type == QuantType::DYNAMIC && isNoopObserver(observer->input(0))) {
-    dequant = insertFP16CastOps(g, observer_out);
-  } else if (
-      quant_type == QuantType::DYNAMIC && !isWeight(module, observer_out)) {
-    Value* dtype = g->insertGetAttr(self, qparam_names.back());
-    std::tie(choose_qparams, quant, dequant) = insertChooseQParamQuantDequant(
-        g, observer_out, dtype, at::Symbol::aten(quantize_func));
-  } else {
-    // Else branch is executed for dynamic weight observers and all observers
-    // for static quant.
-    std::vector<Value*> inputs = {observer_out};
-    // Insert GetAttr nodes for quantization parameters
-    for (const auto& qparam_name : qparam_names) {
-      inputs.push_back(g->insertGetAttr(self, qparam_name));
+
+  // Special case for embedding bag operators
+  if (quant_type == QuantType::DYNAMIC &&
```

Comment:
can we make this branch one of the inner branch of the DYNAMIC branch starting in L318?

Reply:
wait, what is this doing?

Reply:
`insertEmbeddingBagQuantOps` below is run only for the weight tensors. For the indices input we still have an observer attached which we need to disregard. This part of code replaces that observer output with the original value so that the observer can be deleted in cleanup step. I'll update comment to make it more descriptive.

Reply:
why can't we skip inserting observer for input? I remember we implemented that support before for batchnorm?

Reply:
https://codebrowser.bddppq.com/pytorch/pytorch/torch/csrc/jit/passes/quantization/helper.cpp.html#torch::jit::_observe_inputs_call_func

Reply:
I think we spoke offline about this as well but this case is special so we need to observe both. If we don't I run into 
```
RuntimeError: branches for `if` should return values that are observed consistently, if node:%input.22 : Tensor, %weight : Tensor = prim::If(%19) # /home/supriyar/local/miniconda3/envs/pytorch/lib/python3.7/site-packages/torch/nn/functional.py:1908:4
```

Reply:
Oh I see, then could you add a comment for this?